muttlib/dbconn/base.py

```diff
@@ -126,6 +127,10 @@ def __setattr__(self, attr, value):
                 elif attr == 'driver':
                     value = format_drivername(self.dialect, value)
                 attr = 'drivername'
+
+            if version.parse(sqlalchemy_version) >= version.parse("1.4"):
+                self.conn_url = self.conn_url.set(**{attr: value})
+                return self.conn_url
```

Comment:
what is the return of this `.set()` that you are returning?

Reply:
In SQLAlchemy >=1.4, `sqlalchemy.engine.URL` is immutable, so this method returns a modified object.

[Method doc](https://docs.sqlalchemy.org/en/14/core/engines.html#sqlalchemy.engine.URL.set)
[Note on URL immutability](https://docs.sqlalchemy.org/en/14/changelog/migration_14.html#change-5526)